### PR TITLE
sim: Add a config to test libcxx

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
extract from https://github.com/apache/incubator-nuttx/pull/2063, to improve c++ build coverage

## Testing
pass the build

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@yamt do you know how to fix this warning?
```
In file included from libcxx/src/filesystem/directory_iterator.cpp:19:
Error: libcxx/src/filesystem/filesystem_common.h:200:27: error: declaration 'struct timespec' does not declare anything [-Werror]
  200 | using TimeSpec = struct ::timespec;
      |                           ^~~~~~~~
Error: libcxx/src/filesystem/filesystem_common.h:201:24: error: declaration 'struct stat' does not declare anything [-Werror]
  201 | using StatT = struct ::stat;
      |                        ^~~~
cc1plus: all warnings being treated as errors
make[1]: *** [Makefile:87: directory_iterator.o] Error 1
In file included from libcxx/src/filesystem/operations.cpp:20:
Error: libcxx/src/filesystem/filesystem_common.h:200:27: error: declaration 'struct timespec' does not declare anything [-Werror]
  200 | using TimeSpec = struct ::timespec;
      |                           ^~~~~~~~
Error: libcxx/src/filesystem/filesystem_common.h:201:24: error: declaration 'struct stat' does not declare anything [-Werror]
  201 | using StatT = struct ::stat;
      |                        ^~~~
```

File: boards/sim/sim/sim/scripts/Make.defs [hunk 1]
```diff
@@ -59,8 +59,8 @@ ifeq ($(CONFIG_CXX_EXCEPTION),)
   ARCHCPUFLAGSXX += -fno-exceptions
 endif
 ARCHPICFLAGS = -fpic
-ARCHWARNINGS = -Wall -Wstrict-prototypes -Wshadow -Wundef
-ARCHWARNINGSXX = -Wall -Wshadow -Wundef
+ARCHWARNINGS = -Wstrict-prototypes -Wundef
+ARCHWARNINGSXX = -Wundef
```
Comment (yamt) [inline boards/sim/sim/sim/scripts/Make.defs:63]: the removal of -Wshadow is explained in the commit message.
but the removal of -Wall is not. can you explain a bit?

Comment (xiaoxiang781216) [reply boards/sim/sim/sim/scripts/Make.defs:63]: Yes, it isn't required for libc++, but I have saw other 3rd party library can't be the warning free with -Wall, and it's hard to disable -Wall by CFLAGS. Do you hit some special issue without -Wall?

Comment (yamt) [reply boards/sim/sim/sim/scripts/Make.defs:63]: > Yes, it isn't required for libc++, but I have saw other 3rd party library can't be the warning free with -Wall, and it's hard to disable -Wall by CFLAGS.

how is it hard to disable?
-Wall is basically a set of -Wxxx. you can disable each of them by -Wno-xxx.

> Do you hit some special issue without -Wall?

no special issue. i just prefer to enable it as it's often useful.

Comment (xiaoxiang781216) [reply boards/sim/sim/sim/scripts/Make.defs:63]: I mean there isn't -Wno-all to restore to the default.

**Pull request merged**